code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -260,26 +288,67 @@ There are several things that need to be remembered:
 
 	if(client && hud_used)
 		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_EARS) + 1]
-		inv.update_appearance()
+		inv.update_icon()
 
 	if(ears)
-		ears.screen_loc = ui_ears //move the item to the appropriate screen loc
-		if(client && hud_used?.hud_shown)
-			if(hud_used.inventory_shown) //if the inventory is open
-				client.screen += ears //add it to the client's screen
-		update_observer_view(ears,1)
-		overlays_standing[EARS_LAYER] = ears.build_worn_icon(default_layer = EARS_LAYER, default_icon_file = 'icons/mob/clothing/ears.dmi')
-		var/mutable_appearance/ears_overlay = overlays_standing[EARS_LAYER]
-		if(OFFSET_EARS in dna.species.offset_features)
+		var/obj/item/worn_item = ears
+		var/mutable_appearance/ears_overlay
+		update_hud_ears(worn_item)
+
+		var/handled_by_bodytype = TRUE
+		var/icon_file
+
+		if(!(icon_exists(icon_file, RESOLVE_ICON_STATE(worn_item))))
+			handled_by_bodytype = FALSE
+			icon_file = 'icons/mob/clothing/ears.dmi'
+
+		ears_overlay = ears.build_worn_icon(default_layer = EARS_LAYER, default_icon_file = icon_file)
+
+		if(!ears_overlay)
+			return
+		if(OFFSET_EARS in dna.species.offset_features && !handled_by_bodytype)
 			ears_overlay.pixel_x += dna.species.offset_features[OFFSET_EARS][1]
 			ears_overlay.pixel_y += dna.species.offset_features[OFFSET_EARS][2]
 		overlays_standing[EARS_LAYER] = ears_overlay
 	apply_overlay(EARS_LAYER)
-*/
-//SKYRAT EDIT REMOVAL END
 
-//SKYRAT EDIT REMOVAL BEGIN - CUSTOMIZATION (moved to modular)
-/*
+
+/mob/living/carbon/human/update_inv_neck()
+	remove_overlay(NECK_LAYER)
+
+	if(client && hud_used && hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_NECK) + 1])
+		var/atom/movable/screen/inventory/inv = hud_used.inv_slots[TOBITSHIFT(ITEM_SLOT_NECK) + 1]
+		inv.update_icon()
+
+	if(wear_neck)
+		var/obj/item/worn_item = wear_neck
+
+		if(!(ITEM_SLOT_NECK in check_obscured_slots()))
+			var/mutable_appearance/neck_overlay
+			var/icon_file
+			var/handled_by_bodytype = TRUE
+
+			// SKYRAT EDIT ADDITION
+			if(dna.species.bodytype & BODYTYPE_TESHARI)
+				icon_file = TESHARI_NECK_ICON
+			// SKYRAT EDIT END
+
+			if(!(icon_exists(icon_file, RESOLVE_ICON_STATE(worn_item))))
+				handled_by_bodytype = FALSE
+				icon_file = 'icons/mob/clothing/neck.dmi'
```

Comment:
Won't stuff like this be an issue for teshari clothes?

Reply:
Only if the shit doesn't exist